src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeEncoding.cs

```diff
@@ -0,0 +1,99 @@
+        public static Encoding GetEncoding(StringSegment mediaType)
+        {
+            var charset = GetCharsetParameter(mediaType);
+            return GetEncodingFromCharset(charset);
+        }
+
+        /// <summary>
+        /// Gets the <see cref="Encoding"/> for the given <see cref="mediaType"/> if it exists.
+        /// </summary>
+        /// <param name="mediaType">The media type from which to get the charset parameter.</param>
+        /// <returns>The <see cref="Encoding"/> of the media type if it exists or a <see cref="StringSegment"/> without value if not.</returns>
+        public static Encoding GetEncoding(string mediaType)
+        {
+            var charset = GetCharsetParameter(new StringSegment(mediaType));
+            return GetEncodingFromCharset(charset);
+        }
+
+        /// <summary>
+        /// Gets the charset parameter of the given <paramref name="mediaType"/> if it exists.
+        /// </summary>
+        /// <param name="mediaType">The media type from which to get the charset parameter.</param>
+        /// <returns>The charset of the media type if it exists or a <see cref="StringSegment"/> without value if not.</returns>
+        public static StringSegment GetCharsetParameter(StringSegment mediaType)
+        {
+            MediaTypeHeaderValue parsedMediaType = null;
+            if (MediaTypeHeaderValue.TryParse(mediaType.Value, out parsedMediaType))
+            {
+                return new StringSegment(parsedMediaType.Charset);
+            }
+            return new StringSegment();
+        }
+
+        /// <summary>
+        /// Replaces the encoding of the given <paramref name="mediaType"/> with the provided
+        /// <paramref name="encoding"/>.
+        /// </summary>
+        /// <param name="mediaType">The media type whose encoding will be replaced.</param>
+        /// <param name="encoding">The encoding that will replace the encoding in the <paramref name="mediaType"/></param>
+        /// <returns>A media type with the replaced encoding.</returns>
+        public static string ReplaceEncoding(string mediaType, Encoding encoding)
+        {
+            return ReplaceEncoding(new StringSegment(mediaType), encoding);
+        }
+
+        /// <summary>
+        /// Replaces the encoding of the given <paramref name="mediaType"/> with the provided
+        /// <paramref name="encoding"/>.
+        /// </summary>
+        /// <param name="mediaType">The media type whose encoding will be replaced.</param>
+        /// <param name="encoding">The encoding that will replace the encoding in the <paramref name="mediaType"/></param>
+        /// <returns>A media type with the replaced encoding.</returns>
+        public static string ReplaceEncoding(StringSegment mediaType, Encoding encoding)
+        {
+            var parsedMediaType = MediaTypeHeaderValue.Parse(mediaType.Value);
```

Comment:
Why doesn't this return `null` if `mediaType` fails to parse? Doesn't look like `throw`ing is correct for all callers.